src/use_case/room_settings/RoomSettingsInteractor.java

```diff
@@ -0,0 +1,34 @@
+package use_case.room_settings;
+
+import entities.auth.User;
+import entities.rooms.Room;
+import use_case.login.LoginUserDataAccessInterface;
+import use_case.rooms.RoomsDataAccessInterface;
+
+public class RoomSettingsInteractor implements RoomSettingsInputBoundary {
+  private final RoomsDataAccessInterface roomsDataAccessObject;
+  private final LoginUserDataAccessInterface userDataAccessObject;
+
+  private final RoomSettingsOutputBoundary outputBoundary;
+
+  public RoomSettingsInteractor(
+      RoomsDataAccessInterface roomsDataAccessObject,
+      LoginUserDataAccessInterface userDataAccessObject,
+      RoomSettingsOutputBoundary outputBoundary) {
+    this.roomsDataAccessObject = roomsDataAccessObject;
+    this.userDataAccessObject = userDataAccessObject;
+    this.outputBoundary = outputBoundary;
+  }
+
+  @Override
+  public void deleteRoom(User user, Room activeRoom) {
```

Comment:
Does this impact Elasticsearch stuff at all?

Reply:
No, elastic search can only be run from the current room and this removes access to that

Reply:
I see, okay